builtin/v12/migration/miner.go

```diff
@@ -185,14 +220,23 @@ func migrateSectorsWithCache(ctx context.Context, store adt11.Store, cache migra
 			if err != nil {
 				return cid.Undef, xerrors.Errorf("failed to read sectors array: %w", err)
 			}
-			outArray, err := migrateSectorsFromScratch(ctx, store, inArray)
+
+			outArray, outDealHamtCid, err := m.migrateSectorsFromScratch(ctx, store, minerAddr, inArray)
 			if err != nil {
 				return cid.Undef, xerrors.Errorf("failed to migrate sectors from scratch: %w", err)
 			}
 			outRoot, err = outArray.Root()
 			if err != nil {
 				return cid.Undef, xerrors.Errorf("error writing new sectors AMT: %w", err)
 			}
+
+			sectorToDealIdHamtCid = outDealHamtCid
```

Comment:
nit: I would just use one variable name in both if branches to reduce noise.

Reply:
I didn't understand this one -- when i code it what should be the normal way:

```
mikers@mikers-B560-DS3H-AC-Y1:~/dev/filecoin-project/go-state-types$ git diff 
diff --git a/builtin/v12/migration/miner.go b/builtin/v12/migration/miner.go
index 4abacc4..f3216b0 100644
--- a/builtin/v12/migration/miner.go
+++ b/builtin/v12/migration/miner.go
@@ -221,7 +221,7 @@ func (m minerMigrator) migrateSectorsWithCache(ctx context.Context, store adt11.
                                return cid.Undef, xerrors.Errorf("failed to read sectors array: %w", err)
                        }
 
-                       outArray, outDealHamtCid, err := m.migrateSectorsFromScratch(ctx, store, minerAddr, inArray)
+                       outArray, sectorToDealIdHamtCid, err := m.migrateSectorsFromScratch(ctx, store, minerAddr, inArray)
                        if err != nil {
                                return cid.Undef, xerrors.Errorf("failed to migrate sectors from scratch: %w", err)
                        }
@@ -229,8 +229,6 @@ func (m minerMigrator) migrateSectorsWithCache(ctx context.Context, store adt11.
                        if err != nil {
                                return cid.Undef, xerrors.Errorf("error writing new sectors AMT: %w", err)
                        }
-
-                       sectorToDealIdHamtCid = outDealHamtCid
                }
 
                //add to new address sector deal id hamt index
```

i get this compiler error:

builtin/v12/migration/miner.go:224:14: sectorToDealIdHamtCid declared but not used

as if sectorToDealIdHamtCid is a new variable just within the if statement scope.. i found it works if i use a tmp variable but i don't see what the compilers issue is